### PR TITLE
webui: tests: introduce webui code coverage

### PR DESCRIPTION
**Pull request by KKoukiou**
This is not ready, as we are not yet building RPMs with the NODE_ENV=developement scenario.

This will be challenging, see https://github.com/cockpit-project/cockpit-machines/pull/650/files for reference.

### CODE REVIEW

**Comment by KKoukiou** [comment]
Coverage links not working - this needs work.

File: ui/webui/Makefile.am [hunk 1]
```diff
@@ -81,12 +81,13 @@ $(UPDATES_IMG):
...
 integration-test: bots test/common $(UPDATES_IMG)
-	test/run-tests
+	test/run-tests --coverage
 
 # checkout Cockpit's test API; this has no API stability guarantee, so check out a stable tag
 # when you start a new project, use the latest release, and update it from time to time
+# FIXME: workaround for including code coverage support, not yet contained in a released cockpit version
 test/common:
-	git clone -b 266 --depth=1 https://github.com/cockpit-project/cockpit.git tmp/cockpit
+	git clone -b main --depth=1 https://github.com/cockpit-project/cockpit.git tmp/cockpit
```
Comment (M4rtinK) [inline ui/webui/Makefile.am:90]: Is this an intentional change that should be part of the coverage PR ? As (if I understand this correctly) this is essentially unpinning the cockpit tooling version we use for tests I think it should be at least a separate commit.

**Review by M4rtinK — APPROVED** [review]
Looks good to me, but please split the unpinning of cockpit to a separate commit, if intentional. :)

File: ui/webui/Makefile.am (hunk 1 above)
Comment (KKoukiou) [reply ui/webui/Makefile.am:90]: The issue is that testlib changes to suppor tcode coverage is not released yet in cockpit. See the comment above this line. 

This can be jsut bumped up to 267 next week that cockpit releases.

**Comment by KKoukiou** [comment]
This does not work because its a production build.
Code coverage doesn't work with uglified/minified JS.
